daemon/metrics_unix.go

```diff
@@ -0,0 +1,79 @@
+// +build !windows
+
+package daemon
+
+import (
+	"net"
+	"net/http"
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/docker/pkg/mount"
+	"github.com/docker/docker/pkg/plugingetter"
+	"github.com/docker/docker/pkg/plugins"
+	metrics "github.com/docker/go-metrics"
+	"github.com/pkg/errors"
+)
+
+func (d *Daemon) registerMetricsPluginCallback() error {
+	var err error
+	sockPath := filepath.Join(d.configStore.ExecRoot, "metrics.sock")
```

Comment:
Looks like the metrics socket is exposed using the `/metrics` endpoint, irrespective of whether a metrics plugin is available or not. Better to split the function, since `registerMetricsPluginCallback` is doing much more than that.